src/AzureAppConfigurationImpl.ts

```diff
@@ -93,35 +94,38 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
 
         if (options?.refreshOptions?.enabled) {
-            const { watchedSettings, refreshIntervalInMs } = options.refreshOptions;
-            // validate watched settings
-            if (watchedSettings === undefined || watchedSettings.length === 0) {
-                throw new Error("Refresh is enabled but no watched settings are specified.");
+            const { watchedSettings, refreshIntervalInMs, watchAll } = options.refreshOptions;
+            // validate refresh options
+            if (watchAll !== true) {
+                if (watchedSettings === undefined || watchedSettings.length === 0) {
+                    throw new Error("Refresh is enabled but no watched settings are specified.");
+                } else {
+                    for (const setting of watchedSettings) {
+                        if (setting.key.includes("*") || setting.key.includes(",")) {
+                            throw new Error("The characters '*' and ',' are not supported in key of watched settings.");
+                        }
+                        if (setting.label?.includes("*") || setting.label?.includes(",")) {
+                            throw new Error("The characters '*' and ',' are not supported in label of watched settings.");
+                        }
+                        this.#sentinels.push(setting);
+                    }
+                }
+            } else if (watchedSettings && watchedSettings.length > 0) {
+                throw new Error("Watched settings should not be specified when registerAll is enabled.");

```

Comment:
should it be `watchAll`?

Reply:
We will remove `watchAll` from the refreshOptions. See [here](https://github.com/Azure/AppConfiguration-JavaScriptProvider/pull/133#discussion_r1847440695)